sql/sql_select.cc

```diff
@@ -827,8 +827,10 @@ int SELECT_LEX::vers_setup_conds(THD *thd, TABLE_LIST *tables)
       lock_type= TL_READ; // ignore TL_WRITE, history is immutable anyway
     }
 
-    const LEX_CSTRING *fstart= &table->table->vers_start_field()->field_name;
-    const LEX_CSTRING *fend= &table->table->vers_end_field()->field_name;
+    const LEX_CSTRING *fstart=
+        thd->make_clex_string(table->table->vers_start_field()->field_name);
+    const LEX_CSTRING *fend=
+        thd->make_clex_string(table->table->vers_end_field()->field_name);
 
```

Comment:
Could you explain what the bug was? Looking at your fix, I guess it's: `SELECT_LEX::vers_setup_conds` is run only once — on the first `CALL` — it remembers pointers into the table — then the table is closed — you get dangling pointers. Right?

If yes — are you sure you allocate fstart/fend on the correct arena? It must be not the execution arena, but the sp arena that will survive multiple executions.

Reply:
> SELECT_LEX::vers_setup_conds is run only once — on the first CALL

Not really. See another PR https://github.com/MariaDB/server/pull/756

> it remembers pointers into the table — then the table is closed — you get dangling pointers. Right?

Right.

> If yes — are you sure you allocate fstart/fend on the correct arena? It must be not the execution arena, but the sp arena that will survive multiple executions.

It's allocated in statement arena, if needed. See https://github.com/MariaDB/server/blob/10.3/sql/sql_select.cc#L756